### PR TITLE
Add Order model

### PR DESCRIPTION
**Pull request by Lessorg**

### CODE REVIEW

File: src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    @Override
+    public OrderResponseDto placeOrder(User currentUser) {
+
+        Order order = new Order();
+        order.setUser(currentUser);
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(currentUser.getShippingAddress());
+
+        Set<CartItem> cartContent = shoppingCartService.getCartContentByUserId(currentUser.getId());
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java]: Could cartItems be empty in shop cart?

File: src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto placeOrder(User currentUser) {
+
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java]:
```suggestion
```

File: src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import test.project.bookshop.model.OrderItem;
+import test.project.bookshop.model.User;
+import test.project.bookshop.repository.OrderItemRepository;
+import test.project.bookshop.repository.OrderRepository;
+import test.project.bookshop.service.OrderService;
+import test.project.bookshop.service.ShoppingCartService;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java:29]: Think where is `@Transactional` needed

File: src/main/java/test/project/bookshop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -73,8 +74,13 @@ public void delete(Long userId, Long cartItemId) {
         cartItemRepository.deleteById(cartItemId);
     }
 
+    @Override
+    public Set<CartItem> getCartContentByUserId(Long userId) {
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/ShoppingCartServiceImpl.java]: remove

File: src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java (hunk 1 above)
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/OrderServiceImpl.java]: Let's just use here `shoppingCartRepository.findByUserId(userId).orElseThrow(()`

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/test/project/bookshop/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(summary = "Place an order",
+            description = "Place an order for the items in the user's shopping cart")
+    public OrderResponseDto placeOrder(Authentication authentication) {
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/controller/OrderController.java]:
```suggestion
    public OrderResponseDto placeOrder(Authentication authentication, @RequestBody @Valid CreateOrderRequestDto orderDto) {
```

Comment (Elena-Bruyako) [reply src/main/java/test/project/bookshop/controller/OrderController.java]: see an example of request body 

{
  "shippingAddress": "Kyiv, Shevchenko ave, 1"
}

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
check task description and fix comments

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**